core/src/componentset.ts

```diff
@@ -0,0 +1,15 @@
+import { Card } from "./card";
+import { Noble } from "./noble";
+import { Token } from "./token";
+
+export class ComponentSet {
+  cards: Card[];
+  nobles: Noble[];
+  tokens: Token[];
+
+  constructor(cards: Card[], nobles: Noble[], tokens: Token[]) {
```

Comment:
I'm wondering if it's common to write those types twice like this?

Reply:
Hm you mean once for the properties and once for the constructor arguments? Not sure if it's common but I personally think it's a good practice to annotate everything you can.